spec/services/course_search_service_spec.rb

```diff
@@ -65,9 +66,18 @@
         it "orders in descending order" do
           expect(findable_scope).to receive(:select).and_return(inner_query_scope)
           expect(Course).to receive(:where).and_return(findable_scope)
-          expect(findable_scope).to receive(:joins).and_return(select_scope)
-          expect(select_scope).to receive(:select).and_return(order_scope)
-          expect(order_scope).to receive(:order).and_return(expected_scope)
+          expect(findable_scope).to receive(:joins).and_return(joins_provider_scope)
+          expect(joins_provider_scope).to receive(:joins).with(:provider).and_return(select_scope)
+          boosted_distance_column = "
```

Comment:
I wonder if this mocking approach has got a bit out of hand in here now. We're really deep in the implementation now rather than the behaviour. That ship has sailed a bit as there are a lot of existing tests in this pattern.

Reply:
This mocking approach is forcing me to enforce chock points, implementation changes then this test must **break**, dare touch the chock points then be prepared to be ran over. 

It is becoming a little bit of a valueless approach as it has obfuscated the intent, given how much domain knowledge is required to comprehend it, the tests just demonstrates vagueness throughout.

Which was also why I wrote the full blown unit test as it is getting elbow deep. 
So we can have more **fun** picking holes like https://github.com/DFE-Digital/teacher-training-api/pull/1435#discussion_r442320179